codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -1012,6 +1012,15 @@ private V remove0(int h, int i, K name) {
         return value;
     }
 
+    private void remove0(HeaderEntry<K, V> entry) {
+        int i = index(entry.hash);
+        HeaderEntry<K, V> e = entries[i];
+        if (e == entry) {
+            entries[i] = entry.next;
+        }
+        entry.remove();
+    }
```

Comment:
I think you also need to call `--size;`

Can you please also adjust the unit test to check for this ?

Reply:
good catch! fixed.